File: JavaMaps/app/src/main/java/com/mhmmdyzci/javamaps/view/MainActivity.java
package com.mhmmdyzci.javamaps.view;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import androidx.room.Room;

import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;

import com.mhmmdyzci.javamaps.R;
import com.mhmmdyzci.javamaps.adapter.PlaceAdapter;
import com.mhmmdyzci.javamaps.databinding.ActivityMain2Binding;
import com.mhmmdyzci.javamaps.model.Place;
import com.mhmmdyzci.javamaps.roomdb.PlaceDao;
import com.mhmmdyzci.javamaps.roomdb.PlaceDatabase;

import java.util.ArrayList;
import java.util.List;

import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers;
import io.reactivex.rxjava3.disposables.CompositeDisposable;
import io.reactivex.rxjava3.schedulers.Schedulers;

public class MainActivity extends AppCompatActivity {
    private ActivityMain2Binding binding;
    private CompositeDisposable compositeDisposable = new CompositeDisposable();
    PlaceDao placeDao;
    PlaceDatabase db;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMain2Binding.inflate(getLayoutInflater());
        View view = binding.getRoot();

        setContentView(view);
        db = Room.databaseBuilder(getApplicationContext(),PlaceDatabase.class,"Places").allowMainThreadQueries().build();
        placeDao = db.placeDao();
        compositeDisposable.add(placeDao.getall()
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(this::handleResponse)

        );



    }
    private void handleResponse(List<Place> placeList){
        binding.recyclerView.setLayoutManager(new LinearLayoutManager(this));
        PlaceAdapter placeAdapter = new PlaceAdapter(placeList);
        binding.recyclerView.setAdapter(placeAdapter);


    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater menuInflater = getMenuInflater();
        menuInflater.inflate(R.menu.travel_menu,menu);
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        if(item.getItemId()== R.id.add_place){
            Intent intent = new Intent(MainActivity.this, MapsActivity.class);
            intent.putExtra("info","new");
            startActivity(intent);
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        compositeDisposable.clear();
    }


}

File: ViewBindingKotlin/app/src/main/java/com/mhmmdyzci/viewbindingkotlin/MainActivity.kt
package com.mhmmdyzci.viewbindingkotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.TextView
import com.mhmmdyzci.viewbindingkotlin.databinding.ActivityMainBinding


class MainActivity : AppCompatActivity() {
    //private lateinit var textView: TextView
    // Lateinit var -> seni daha sonra tanımlıcam
    private  lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)

        //textView = findViewById(R.id.textView)

    }
    fun changeName(view : View){
        //textView.text = "Merhaba kotlin"
        /*val textView = findViewById<TextView>(R.id.textView)
        textView.text="Merhaba kotlin"
         */
        binding.textView.text = "Merhaba kotlin"


    }
}